drivers/flash/soc_flash_nrf.h

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __SOC_FLASH_NRF_H__
+#define __SOC_FLASH_NRF_H__
+
+#include <kernel.h>
+
+#define FLASH_OP_DONE    (0) /* 0 for compliance with the driver API. */
+#define FLASH_OP_ONGOING (-1)
+
+struct flash_context {
+	uint32_t data_addr;  /* Address of data to write. */
+	uint32_t flash_addr; /* Address of flash to write or erase. */
+	uint32_t len;        /* Size of data to write or erase [B]. */
+#ifndef CONFIG_SOC_FLASH_NRF_RADIO_SYNC_NONE
+	uint8_t  enable_time_limit; /* set execution limited to the execution
+				     * window.
+				     */
+#endif /* !CONFIG_SOC_FLASH_NRF_RADIO_SYNC_NONE */
+#if defined(CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE)
+	uint32_t flash_addr_next;
+#endif /* CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE */
+}; /*< Context type for f. @ref write_op @ref erase_op */
+
+#ifndef CONFIG_SOC_FLASH_NRF_RADIO_SYNC_NONE
+
+#if defined(CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE)
+/* The timeout is multiplied by 1.5 because switching tasks may take
+ * significant portion of time.
+ */
+#define FLASH_TIMEOUT_MS ((FLASH_PAGE_ERASE_MAX_TIME_US) * \
+			  (FLASH_PAGE_MAX_CNT) / 1000 * 15 / 10)
+#else
+
+#define FLASH_TIMEOUT_MS ((FLASH_PAGE_ERASE_MAX_TIME_US) * \
+			  (FLASH_PAGE_MAX_CNT) / 1000)
+#endif /* CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE */
+
+/**
+ * @defgroup nrf_flash_sync sync backend API
+ *
+ * API declared below contains prototypes of function which shall be
+ * implemented by the synchronization backend.
+ * @{
+ */
+
+/**
+ * Callback which executes the flash operation.
+ *
+ * @param context pointer to flash_context structure.
+ * @retval @ref FLASH_OP_DONE once operation was done, @ref FLASH_OP_ONGOING if
+ *         operation needs more time for execution.
+ */
+typedef int (*flash_op_handler_t) (void *context);
+
+struct flash_op_desc {
+	flash_op_handler_t handler;
+	struct flash_context *context; /* [in,out] */
+};
+
+/**
+ * Synchronization backend driver initialization procedure.
+ *
+ * This will be run within flash driver initialization
+ */
+int nrfsb_sync_backend_init(void);
```

Comment:
this is not a good prefix. Can we use something like `nrf_flash_sync_*` on all calls instead, and remove the "backend" word?
```
nrf_flash_sync_init()
nrf_flash_sync_set_context()
nrf_flash_sync_is_required();
nrf_flash_sync_exe() / nrf_flash_sync_op()
```